storage/remote/queue_manager.go

```diff
@@ -514,9 +562,9 @@ func (s *shards) sendSamples(samples model.Samples) {
 }
 
 // sendSamples to the remote storage with backoff for recoverable errors.
-func (s *shards) sendSamplesWithBackoff(samples model.Samples) {
+func (s *shards) sendSamplesWithBackoff(samples []tsdb.RefSample) {
 	backoff := s.qm.cfg.MinBackoff
-	req := ToWriteRequest(samples)
+	req := toWriteRequest(samples, s.qm.series)
```

Comment:
Should access to s.qm.series be locked?  And won't this introduce contention?  Perhaps each shard should have it own series map?